addons/point_of_sale/static/src/app/models/utils/recursive_serialization.js

```diff
@@ -184,9 +184,11 @@ const deepSerialization = (
             continue;
         }
         if (fieldName === "id") {
-            if (typeof record[fieldName] === "number") {
-                result[fieldName] = record[fieldName];
+            let value = record[fieldName];
+            if (typeof value === "string") {
+                value = parseInt(value.split("_")[1]);
             }
+            result[fieldName] = value !== undefined ? value : false;
```

Comment:
sorry but the `number` that can be extracted in the `string` local id is not really a db ID, it's just a local count. This approach may resolve the traceback but it won't really solve the problem.